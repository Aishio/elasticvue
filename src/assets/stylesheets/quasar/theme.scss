.body--light {
  .bg-darken {
    color: #000;
    background-color: #F5F5F5;
  }

  .q-banner.bg-grey-7 {
    color: #fff;
  }
}

.body--dark {
  .bg-darken {
    color: #fff;
    background-color: #272727;
  }

  .q-btn-group {
    background-color: #111;

    .q-btn {
      background-color: #333;
    }

    .q-btn:first-child {
      margin-right: 2px;
    }
  }

  .q-table__control {
    .q-btn:not(:disabled) {
      color: #fff !important;
    }
  }
}
