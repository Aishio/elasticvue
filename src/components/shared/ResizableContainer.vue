<template>
  <div>
    <div :style="style" ref="resizedWrapper">
      <slot></slot>
    </div>

    <v-flex right d-inline-flex my-2>
      <btn-group small>
        <v-btn flat
               ref="decreaseHeightButton"
               :disabled="!canDecreseHeight"
               title="Decrease height"
               @click.native="decreaseHeight">
          <v-icon>arrow_upward</v-icon>
        </v-btn>
        <v-btn flat
               ref="increaseHeightButton"
               title="Increase height"
               @click.native="increaseHeight">
          <v-icon>arrow_downward</v-icon>
        </v-btn>
        <v-btn flat
               class="v-btn--dense ma-0"
               title="Resize"
               @mousedown="onDragStart">
          <v-icon>vertical_align_center</v-icon>
        </v-btn>
      </btn-group>
    </v-flex>
  </div>
</template>

<script>
  import BtnGroup from '@/components/shared/BtnGroup'

  export default {
    name: 'resizable-container',
    props: {
      initialWidth: {
        default: '100%',
        type: String
      },
      initialHeight: {
        default: 350,
        type: Number
      },
      stepSize: {
        default: 150,
        type: Number
      }
    },
    data () {
      return {
        width: this.initialWidth,
        height: this.initialHeight,
        dragStartY: 0,
        dragStartHeight: this.initialHeight,
        resizing: false
      }
    },
    computed: {
      style () {
        return `width: ${this.width}; height: ${this.height}px`
      },
      canDecreseHeight () {
        return (this.height - this.stepSize) >= this.initialHeight
      }
    },
    methods: {
      increaseHeight () {
        this.height += this.stepSize
        this.triggerResizeAndFocusRef('increaseHeightButton')
      },
      decreaseHeight () {
        if (this.canDecreseHeight) this.height -= this.stepSize
        this.triggerResizeAndFocusRef('decreaseHeightButton')
      },
      triggerResizeAndFocusRef (ref) {
        this.$nextTick(() => {
          if (window !== undefined) {
            this.triggerResize()
          }
          this.$refs[ref].$el.focus()
        })
      },
      triggerResize () {
        window.dispatchEvent(new Event('resize'))
      },
      onDragStart (e) {
        this.resizing = true
        this.dragStartY = e.pageY
        this.dragStartHeight = this.height
      },
      onDrag (e) {
        if (this.resizing) {
          const newHeight = this.dragStartHeight + e.pageY - this.dragStartY
          if (newHeight > this.initialHeight) this.height = newHeight
        }
      },
      onDragEnd () {
        this.resizing = false
        this.dragStartY = 0
        this.$nextTick(() => {
          if (window !== undefined) {
            this.triggerResize()
          }
        })
      }
    },
    mounted () {
      if (window !== undefined) {
        window.addEventListener('mouseup', this.onDragEnd)
        window.addEventListener('mousemove', this.onDrag)
      }
    },
    destroyed () {
      if (window !== undefined) {
        window.removeEventListener('mouseup', this.onDragEnd)
        window.removeEventListener('mousemove', this.onDrag)
      }
    },
    components: {
      BtnGroup
    }
  }
</script>
