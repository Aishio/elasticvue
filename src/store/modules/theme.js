export const theme = {
  state: {
    dark: true
  },
  mutations: {
    setTheme (state, flag) {
      state.dark = flag
    }
  }
}
