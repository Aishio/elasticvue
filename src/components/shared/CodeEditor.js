const Brace = import('brace')
import('brace/mode/json')
import('brace/theme/monokai')

export default {
  data () {
    return {
      editor: null
    }
  },
  props: {
    value: {
      default: () => {
        return {}
      },
      type: [Object, String]
    },
    readOnly: {
      type: Boolean,
      default: false
    },
    externalHandler: {
      type: Function,
      default: () => {
      }
    },
    useWorker: {
      type: Boolean,
      default: true
    }
  },
  computed: {
    darkTheme () {
      return this.$store.state.theme.dark
    }
  },
  methods: {
    setTheme (dark) {
      if (dark) {
        this.editor.setTheme('ace/theme/monokai')
      } else {
        this.editor.setTheme('ace/theme/textmate')
      }
    },
    setEditorValue (value) {
      if (typeof value === 'string') {
        this.editor.setValue(value, 1)
      } else {
        this.editor.setValue(JSON.stringify(value, null, '\t'), 1)
      }
    },
    setupAceEditor () {
      this.editor = Brace.edit(this.$el, {autoScrollEditorIntoView: true})
      this.$emit('init', this.editor)
      this.editor.getSession().setUseWorker(this.useWorker)
      this.editor.getSession().setMode('ace/mode/json')
      this.editor.setFontSize('14px')
      this.editor.$blockScrolling = Infinity
      this.setTheme(this.$store.state.theme.dark)
    }
  },
  watch: {
    darkTheme (value) {
      this.setTheme(value)
    },
    value (value) {
      if (this.readOnly) this.setEditorValue(value)
    }
  },
  mounted () {
    this.setupAceEditor()

    this.setEditorValue(this.value)
    if (this.readOnly) {
      this.editor.setReadOnly(true)
      this.editor.setShowPrintMargin(false)
    } else {
      this.editor.commands.addCommand({
        name: 'externalHandler',
        bindKey: {win: 'Ctrl+ENTER', mac: 'Command+ENTER', linux: 'Ctrl+ENTER'},
        exec: this.externalHandler
      })
    }

    this.editor.on('change', () => {
      this.$emit('input', this.editor.getValue())
    })
  },
  beforeDestroy () {
    this.editor.destroy()
    this.editor.container.remove()
  },
  render: function (createElement) {
    return createElement('div', {attrs: {style: 'height: 100%; width: 100%'}, class: 'bordered'})
  }
}
