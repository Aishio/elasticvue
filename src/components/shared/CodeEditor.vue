<template>
  <div class="bordered code-editor">
    <div ref="editor" style="height: 100%; width: 100%"/>

    <div class="code-editor__actions pa-2">
      <v-btn v-if="!readOnly"
             :disabled="!valid"
             small
             title="Beautify (Ctrl+ALT+L)"
             class="vertical-align--top mr-2"
             @click="beautify">
        <v-icon small>mdi-code-braces</v-icon>
      </v-btn>

      <v-btn-toggle v-model="settings" multiple>
        <v-btn v-if="!readOnly" text small title="Use 2 spaces for beautify (instead of tabs)">
          <v-icon small>mdi-keyboard-space</v-icon>
        </v-btn>
        <v-btn text small title="Wrap lines">
          <v-icon small>mdi-wrap</v-icon>
        </v-btn>
      </v-btn-toggle>
    </div>
  </div>
</template>

<script>
  import Brace from 'brace'
  import 'brace/mode/json'
  import 'brace/theme/monokai'
  import 'brace/ext/searchbox'

  export default {
    props: {
      value: {
        type: null, // any
        default: ''
      },
      readOnly: {
        type: Boolean,
        default: false
      },
      externalCommands: {
        type: Array,
        default: () => {
          return []
        }
      },
      useWorker: {
        type: Boolean,
        default: false
      }
    },
    data () {
      return {
        editor: null,
        settings: []
      }
    },
    computed: {
      darkTheme () {
        return this.$store.state.theme.dark
      },
      wrapLines () {
        return this.settings.includes(1)
      },
      valid () {
        if (this.value === '') return true
        try {
          (JSON.parse(this.value))
          return true
        } catch (error) {
          return false
        }
      }
    },
    watch: {
      darkTheme (value) {
        this.setTheme(value)
      },
      wrapLines (value) {
        this.editor.getSession().setUseWrapMode(value)
      },
      value (value) {
        if (this.editor.getValue() !== value) {
          this.setEditorValue(value)
        }
      },
      settings (value) {
        if (value.includes(0)) {
          this.editor.getSession().setTabSize(2)
          this.editor.getSession().setUseSoftTabs(true)
        } else {
          this.editor.getSession().setTabSize(4)
          this.editor.getSession().setUseSoftTabs(false)
          this.beautify()
        }
      }
    },
    mounted () {
      this.setupAceEditor()

      this.setEditorValue(this.value)
      if (this.readOnly) {
        this.editor.setReadOnly(true)
      } else {
        this.editor.commands.addCommands(this.externalCommands)
        this.editor.commands.addCommand({
          bindKey: { win: 'Ctrl+Alt+L', mac: 'Command+Alt+L', linux: 'Ctrl+Alt+L' },
          exec: this.beautify
        })
      }

      /* istanbul ignore next */
      this.editor.on('change', () => {
        this.$emit('input', this.editor.getValue())
      })
    },
    beforeDestroy () {
      this.editor.destroy()
      this.editor.container.remove()
    },
    methods: {
      beautify () {
        if (this.value) {
          let newValue = this.stringifyJson(JSON.parse(this.editor.getValue()))
          this.$emit('input', newValue)
        }
      },
      setTheme (dark) {
        if (dark) {
          this.editor.setTheme('ace/theme/monokai')
        } else {
          this.editor.setTheme('ace/theme/textmate')
        }
      },
      setEditorValue (value) {
        if (typeof value === 'string') {
          this.editor.setValue(value, 1)
        } else {
          this.editor.setValue(this.stringifyJson(value), 1)
        }
      },
      setupAceEditor () {
        this.editor = Brace.edit(this.$refs.editor, { autoScrollEditorIntoView: true })
        this.$emit('init', this.editor)
        this.editor.getSession().setUseWorker(this.readOnly ? false : this.useWorker)
        this.editor.getSession().setMode('ace/mode/json')
        this.editor.getSession().setUseWrapMode(this.wrapLines)
        this.editor.setFontSize('13px')
        this.editor.setShowPrintMargin(false)
        this.editor.$blockScrolling = Infinity
        this.setTheme(this.$store.state.theme.dark)
      },
      stringifyJson (value) {
        let useSpaces = this.settings.includes(0)
        return JSON.stringify(value, null, useSpaces ? '  ' : '\t')
      }
    }
  }
</script>
