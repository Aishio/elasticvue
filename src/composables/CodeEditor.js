import { queryKeywords, querySnippets, queryValues } from '../autocomplete'

import ace from 'ace-builds/src/ace'

import 'ace-builds/src/theme-tomorrow_night'
import 'ace-builds/src/mode-json'
import 'ace-builds/src/ext-searchbox'
import 'ace-builds/src/ext-language_tools'
import { useThemeStore } from '../store/theme'
import { onBeforeUnmount, onMounted, ref, watch } from 'vue'
import { useCodeEditorStore } from '../store/code_editor'
import { beautify } from '../helpers/beautify'
import { writeToClipboard } from '../helpers/clipboard'

export const useCodeEditor = (editorRef, { readOnly, focus, initialValue, emit }) => {
  let aceEditor = null
  let aceEditorSession = null

  const themeStore = useThemeStore()
  const codeEditorStore = useCodeEditorStore()

  onMounted(() => {
    setupAceEditor(editorRef.value, { readOnly })
    if (!readOnly) setupEditingUtils()
    setWrapLines(codeEditorStore.wrapLines)
    setWhitespace(codeEditorStore.useSpaces)
    setTheme(themeStore.dark)
    if (initialValue.value) {
      const beautifulValue = beautify(initialValue.value, codeEditorStore.useSpaces)
      setEditorValue(beautifulValue)
    }
    if (focus) aceEditor.focus()
  })

  watch(() => initialValue.value, newValue => {
    if (aceEditor.getValue() !== newValue) {
      const beautifulValue = beautify(newValue, true)
      setEditorValue(beautifulValue)
    }
  })

  const setTheme = dark => {
    if (dark) {
      aceEditor.setTheme('ace/theme/tomorrow_night')
    } else {
      aceEditor.setTheme('ace/theme/textmate')
    }
  }

  watch(() => themeStore.dark, newValue => (setTheme(newValue)))

  const setWhitespace = useSpaces => {
    if (useSpaces) {
      aceEditorSession.setTabSize(2)
      aceEditorSession.setUseSoftTabs(true)
    } else {
      aceEditorSession.setTabSize(4)
      aceEditorSession.setUseSoftTabs(false)
    }
  }

  const setWrapLines = wrapLines => {
    aceEditorSession.setUseWrapMode(wrapLines)
  }

  const setupAceEditor = (editorRef, additionalOptions) => {
    aceEditor = ace.edit(editorRef, {
      autoScrollEditorIntoView: true
    })
    aceEditorSession = aceEditor.getSession()
    aceEditorSession.setUseWorker(false)
    aceEditorSession.setMode('ace/mode/json')
    aceEditor.setFontSize('13px')
    aceEditor.setShowPrintMargin(false)
    aceEditor.$blockScrolling = Infinity
    if (additionalOptions) aceEditor.setOptions(additionalOptions)
  }

  const setupEditingUtils = () => {
    aceEditor.commands.addCommand({
      bindKey: { win: 'Ctrl+Alt+L', mac: 'Command+Alt+L', linux: 'Ctrl+Alt+L' },
      exec: () => {
        beautifyEditorValue()
      }
    })

    const { completer } = initializeSnippets()
    aceEditor.setOptions({
      enableBasicAutocompletion: [completer],
      enableLiveAutocompletion: true,
      enableSnippets: true
    })

    if (emit) {
      aceEditor.on('change', () => {
        emit('update:modelValue', aceEditor.getValue())
      })
    }
  }

  const unmountEditor = () => {
    aceEditor.destroy()
    aceEditor.container.remove()
  }
  onBeforeUnmount(unmountEditor)

  const copyContent = () => {
    writeToClipboard(aceEditor.getValue())
  }

  watch(() => codeEditorStore.wrapLines, setWrapLines)
  watch(() => codeEditorStore.useSpaces, newValue => {
    setWhitespace(newValue)
    beautifyEditorValue()
  })

  const setEditorValue = value => {
    aceEditor.setValue(value, 1)
  }

  const beautifyEditorValue = () => {
    const newValue = beautify(aceEditor.getValue(), codeEditorStore.useSpaces)
    setEditorValue(newValue)
  }

  return {
    copyContent,
    beautifyEditorValue
  }
}

export const initializeSnippets = () => {
  let snippetManager

  ace.config.loadModule('ace/snippets', m => {
    snippetManager = m.snippetManager
    const parsedSnippets = snippetManager.parseSnippetFile(querySnippets)
    snippetManager.register(parsedSnippets, 'json')
  })

  const snippets = snippetManager.snippetMap.json.map(snip => {
    return {
      caption: snip.name,
      snippet: snip.content,
      meta: '[Snip]',
      type: 'snippet',
      score: 1,
      docHTML: snip.content
    }
  })

  const completer = {
    getCompletions: function (editor, session, pos, prefix, callback) {
      const token = session.getTokenAt(pos.row, pos.col)
      const row = session.getLine(pos.row)
      const addQuotes = token && token.type === 'text' && token.value.indexOf('"') === -1
      const isKey = row && row.indexOf('":') === -1

      const keywords = queryKeywords.map(word => {
        let actualWord = word
        if (addQuotes && isKey) {
          actualWord = `"${word}": `
        } else if (addQuotes) {
          actualWord = `"${word}"`
        }

        return {
          caption: word,
          value: actualWord
        }
      })

      const values = queryValues.map(word => {
        if (addQuotes) {
          return {
            caption: word,
            value: `"${word}"`
          }
        } else {
          return {
            caption: word,
            value: word
          }
        }
      })

      let allCompletions = []

      if (isKey && addQuotes) allCompletions = snippets
      if (isKey) allCompletions = allCompletions.concat(keywords)
      if (!isKey) allCompletions = allCompletions.concat(values)

      return callback(null, allCompletions)
    }
  }

  return {
    completer
  }
}
