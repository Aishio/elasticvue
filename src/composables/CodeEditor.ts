import { queryKeywords, querySnippets, queryValues } from '../autocomplete'

// @ts-ignore
import ace from 'ace-builds/src-noconflict/ace'

// @ts-ignore
ace.config.setModuleLoader('ace/mode/json', () => import('ace-builds/src-noconflict/mode-json.js'))
// @ts-ignore
ace.config.setModuleLoader('ace/theme/tomorrow_night', () => import('ace-builds/src-noconflict/theme-tomorrow_night.js'))
// @ts-ignore
ace.config.setModuleLoader('ace/ext/searchbox', () => import('ace-builds/src-noconflict/ext-searchbox.js'))
// @ts-ignore
ace.config.setModuleLoader('ace/ext/language_tools', () => import('ace-builds/src-noconflict/ext-language_tools.js'))
// @ts-ignore
ace.config.setModuleLoader('ace/snippets', () => import('ace-builds/src-noconflict/ext-language_tools.js'))

import { onBeforeUnmount, onMounted, Ref, watch } from 'vue'
import { useThemeStore } from '../store/theme'
import { useCodeEditorStore } from '../store/codeEditor.ts'
import { beautify } from '../helpers/beautify'
import { writeToClipboard } from '../helpers/clipboard'

type Completion = SimpleCompletion | AceAjax.Completion
type SimpleCompletion = {
  value: string,
  caption: string
}

export const useCodeEditor = (editorRef: Ref<HTMLElement | null>, {
  readOnly,
  initialValue,
  emit,
  commands
}: {
  readOnly: boolean,
  initialValue: Ref<string>,
  emit?: any,
  commands?: AceAjax.EditorCommand[]
}) => {
  let aceEditor: AceAjax.Editor = <AceAjax.Editor>{}
  let aceEditorSession: AceAjax.IEditSession = <AceAjax.IEditSession>{}

  const themeStore = useThemeStore()
  const codeEditorStore = useCodeEditorStore()

  onMounted(() => {
    if (!editorRef.value) return
    setupAceEditor(editorRef.value, { readOnly })
    if (!readOnly) setupEditingUtils()
    setWrapLines(codeEditorStore.wrapLines)
    setWhitespace(codeEditorStore.useSpaces)
    setTheme(themeStore.dark)
    if (initialValue.value) {
      const beautifulValue = beautify(initialValue.value, codeEditorStore.useSpaces)
      setEditorValue(beautifulValue)
    }
  })

  watch(() => initialValue.value, newValue => {
    if (aceEditor.getValue() !== newValue) {
      const beautifulValue = beautify(newValue, true)
      setEditorValue(beautifulValue)
    }
  })

  const setTheme = (dark: boolean) => {
    if (dark) {
      aceEditor.setTheme('ace/theme/tomorrow_night')
    } else {
      aceEditor.setTheme('ace/theme/textmate')
    }
  }

  watch(() => themeStore.dark, newValue => (setTheme(newValue)))

  const setWhitespace = (useSpaces: boolean) => {
    if (useSpaces) {
      aceEditorSession.setTabSize(2)
      aceEditorSession.setUseSoftTabs(true)
    } else {
      aceEditorSession.setTabSize(4)
      aceEditorSession.setUseSoftTabs(false)
    }
  }

  const setWrapLines = (wrapLines: boolean) => {
    aceEditorSession.setUseWrapMode(wrapLines)
  }

  const setupAceEditor = (editorRef: HTMLElement, additionalOptions: object) => {
    aceEditor = ace.edit(editorRef, {
      autoScrollEditorIntoView: true
    })
    aceEditorSession = aceEditor.getSession()
    aceEditorSession.setUseWorker(false)
    aceEditorSession.setMode('ace/mode/json')
    aceEditor.setShowPrintMargin(false)
    // @ts-ignore
    aceEditor.$blockScrolling = Infinity
    aceEditor.setOptions({ fontFamily: 'Hack', fontSize: '14px' })
    if (additionalOptions) aceEditor.setOptions(additionalOptions)
    if (commands) aceEditor.commands.addCommands(commands)
  }

  const setupEditingUtils = () => {
    aceEditor.commands.addCommand({
      bindKey: { win: 'Ctrl+Alt+L', mac: 'Command+Alt+L' },
      exec: () => {
        beautifyEditorValue()
      }
    })

    initializeSnippets(aceEditor)

    if (emit) {
      aceEditor.on('change', () => {
        emit('update:modelValue', aceEditor.getValue())
      })
    }
  }

  const unmountEditor = () => {
    aceEditor.destroy()
    aceEditor.container.remove()
  }
  onBeforeUnmount(unmountEditor)

  const copyContent = () => {
    writeToClipboard(aceEditor.getValue())
  }

  watch(() => codeEditorStore.wrapLines, setWrapLines)
  watch(() => codeEditorStore.useSpaces, newValue => {
    setWhitespace(newValue)
    beautifyEditorValue()
  })

  const setEditorValue = (value: string) => (aceEditor.setValue(value, 1))

  const beautifyEditorValue = () => {
    const value = aceEditor.getValue()
    if (!value) return
    const newValue = beautify(value, codeEditorStore.useSpaces)
    setEditorValue(newValue)
  }

  return {
    copyContent,
    beautifyEditorValue
  }
}

export const initializeSnippets = (aceEditor: AceAjax.Editor) => {
  ace.config.loadModule('ace/snippets', () => {
    const { snippetManager } = ace.require('ace/snippets')
    const parsedSnippets = snippetManager.parseSnippetFile(querySnippets)
    snippetManager.register(parsedSnippets, 'json')

    const snippets: AceAjax.Completion[] = snippetManager.snippetMap.json.map((snip: Record<string, any>) => {
      return {
        caption: snip.name,
        snippet: snip.content,
        meta: '[Snip]',
        type: 'snippet',
        score: 1,
        docHTML: snip.content
      }
    })

    const completer = {
      getCompletions: function (_editor: AceAjax.Editor, session: AceAjax.IEditSession, pos: AceAjax.Position, _prefix: string, callback: any) {
        const token = session.getTokenAt(pos.row, pos.column)
        const row = session.getLine(pos.row)
        const addQuotes = token && token.type === 'text' && token.value.indexOf('"') === -1
        const isKey = row && row.indexOf('":') === -1

        const keywords = queryKeywords.map(word => {
          let actualWord = word
          if (addQuotes && isKey) {
            actualWord = `"${word}": `
          } else if (addQuotes) {
            actualWord = `"${word}"`
          }

          return {
            caption: word,
            value: actualWord
          }
        })

        const values = queryValues.map(word => {
          if (addQuotes) {
            return {
              caption: word,
              value: `"${word}"`
            }
          } else {
            return {
              caption: word,
              value: word
            }
          }
        })

        let allCompletions: Completion[] = []

        if (isKey && addQuotes) allCompletions = snippets
        if (isKey) allCompletions = allCompletions.concat(keywords)
        if (!isKey) allCompletions = allCompletions.concat(values)

        return callback(null, allCompletions)
      }
    }

    aceEditor.setOptions({
      enableBasicAutocompletion: [completer],
      enableLiveAutocompletion: true,
      enableSnippets: true
    })
  })
}
