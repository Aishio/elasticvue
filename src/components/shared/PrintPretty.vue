<template>
  <v-flex py-2>
    <resizable-container :initial-height="initialHeight">
      <code-editor :value="document" :use-worker="false" read-only/>
    </resizable-container>
  </v-flex>
</template>

<script>
  import ResizableContainer from '@/components/shared/ResizableContainer'
  import Loading from '@/components/shared/Loading'

  export default {
    name: 'PrintPretty',
    components: {
      'code-editor': () => ({
        component: import(/* webpackChunkName: "code-editor" */ '@/components/shared/CodeEditor'),
        loading: Loading
      }),
      ResizableContainer
    },
    props: {
      document: {
        default: () => {
          return {}
        },
        type: [Object, Array]
      },
      initialHeight: {
        default: 600,
        type: Number
      }
    }
  }
</script>
