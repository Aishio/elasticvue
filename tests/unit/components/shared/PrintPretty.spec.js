import Vuex from 'vuex'
import Vuetify from 'vuetify'
import VBtn from 'vuetify/es5/components/VBtn'
import VIcon from 'vuetify/es5/components/VIcon'
import VGrid from 'vuetify/es5/components/VGrid'
import CodeEditor from '@/components/shared/CodeEditor'
import ResizableContainer from '@/components/shared/ResizableContainer'
import PrintPretty from '@/components/shared/PrintPretty'
import { createLocalVue, mount } from '@vue/test-utils'
import { theme } from '@/store/modules/theme'
import '../../mocks/codeEditorMocks'

describe('components/shared/PrintPretty.vue', () => {
  let localVue
  let store

  beforeEach(() => {
    localVue = createLocalVue()

    localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(Vuetify, {
      components: {
        CodeEditor,
        ResizableContainer,
        VBtn,
        VIcon,
        VGrid
      }
    })
    store = new Vuex.Store({
      modules: {theme}
    })
  })

  it('should render correct default contents', () => {
    const wrapper = mount(PrintPretty, {localVue, store})
    expect(wrapper.find('div > div').element.style.height).toEqual(`${wrapper.vm.initialHeight}px`)
  })

  it('correctly sets the initialHeight of its resizableContainer', () => {
    const height = 250
    const wrapper = mount(PrintPretty, {
      localVue: localVue,
      store: store,
      propsData: {
        initialHeight: height
      }
    })
    expect(wrapper.find('div > div').element.style.height).toEqual(`${height}px`)
  })
})
