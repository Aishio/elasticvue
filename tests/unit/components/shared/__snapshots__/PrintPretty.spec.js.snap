// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shared/PrintPretty.vue correctly sets the initialHeight of its resizableContainer 1`] = `
<div>
  <v-row>
    <v-col
      class="my-0"
      cols="12"
      md="9"
    >
      <!---->
    </v-col>
     
    <v-col
      class="my-0 mx-2"
      cols="12"
      md="3"
    >
      <v-btn-toggle
        class="right"
      >
        <v-btn
          title="Wrap lines"
          value="true"
        >
          <v-icon>
            mdi-wrap
          </v-icon>
        </v-btn>
      </v-btn-toggle>
    </v-col>
  </v-row>
   
  <v-col
    class="py-2"
  >
    <div
      class="resizable-container"
      style="width: 100%; height: 250px;"
    >
      <div
        class="bordered ace_editor ace-monokai ace_dark"
        style="height: 100%; width: 100%; font-size: 14px;"
      >
        <textarea
          autocapitalize="off"
          autocorrect="off"
          class="ace_text-input"
          spellcheck="false"
          style="opacity: 0;"
          wrap="off"
        />
        <div
          aria-hidden="true"
          class="ace_gutter"
        >
          <div
            class="ace_layer ace_gutter-layer ace_folding-enabled"
          />
          <div
            class="ace_gutter-active-line"
          />
        </div>
        <div
          class="ace_scroller"
        >
          <div
            class="ace_content"
          >
            <div
              class="ace_layer ace_print-margin-layer"
            >
              <div
                class="ace_print-margin"
                style="left: 4px; visibility: hidden;"
              />
            </div>
            <div
              class="ace_layer ace_marker-layer"
            />
            <div
              class="ace_layer ace_text-layer"
              style="padding: 0px 4px;"
            />
            <div
              class="ace_layer ace_marker-layer"
            />
            <div
              class="ace_layer ace_cursor-layer ace_hidden-cursors"
            >
              <div
                class="ace_cursor"
              />
            </div>
          </div>
        </div>
        <div
          class="ace_scrollbar ace_scrollbar-v"
          style="display: none; width: 20px;"
        >
          <div
            class="ace_scrollbar-inner"
            style="width: 20px;"
          />
        </div>
        <div
          class="ace_scrollbar ace_scrollbar-h"
          style="display: none; height: 20px;"
        >
          <div
            class="ace_scrollbar-inner"
            style="height: 20px;"
          />
        </div>
        <div
          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
        >
          <div
            style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
          />
          <div
            style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
          >
            XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
          </div>
        </div>
      </div>
       
      <div
        class="resizable-container__vertical-handler"
      />
    </div>
  </v-col>
</div>
`;

exports[`components/shared/PrintPretty.vue should render correct default contents 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="my-0 col-md-9 col-12"
    >
      <!---->
    </div>
     
    <div
      class="my-0 mx-2 col-md-3 col-12"
    >
      <div
        class="right v-item-group theme--light v-btn-toggle"
      >
        <button
          class="v-btn v-item--active v-btn--active v-btn--contained theme--light v-size--default"
          title="Wrap lines"
          type="button"
          value="true"
        >
          <span
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-wrap theme--light"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
   
  <div
    class="py-2 col"
  >
    <div
      class="resizable-container"
      style="width: 100%; height: 600px;"
    >
      <!---->
       
      <div
        class="resizable-container__vertical-handler"
      />
    </div>
  </div>
</div>
`;
